conda/common/configuration.py

```diff
@@ -210,9 +210,9 @@ def value(self, parameter_obj):
             # TODO: add stripping of !important, !top, and !bottom
             raw_value = self._raw_value
             if string_delimiter in raw_value:
```

Comment:
`self._raw_value` is always a `str`, right?
`str.split` always returns a list of strings so that you can simplify the whole if-else-block to
```python
value = [v for v in self._raw_value.split(string_delimiter) if v]
```
. If you want to allow empty strings anywhere in the sequence, you could use a trailing separator to distinguish `""`->`[]` from `","`->`[""]` with
```python
value = self._raw_value.split(string_delimiter)  # str.split always returns a list with length >=1
value = value if value[-1] else value[:-1]
```

Reply:
> self._raw_value is always a str, right?

I guess in this case that's safe to assume.  I feel like I'd want to add an `assert` in there though just to be safe.

> If you want to allow empty strings anywhere in the sequence

Can you think of a use-case that we currently have, or one that we could potentially have, where you'd ever want an empty string?  I can't, at the moment.

Reply:
The currently available `SequenceParameter`s in `base/context.py` are lists of packages (specs), features, paths (/subdirs), or channels.
So, no real use case there. Hence this would be more of a "possible forward-compatibility" decision.

Reply:
I say for now that we don't support empty strings.  I can see it just as easily causing us problems in the future, as it could saving us from a future problem.

Reply:
That's fine by me. My mind is oscillating between 30/70, 50/50, and 70/30 on whether the one or the other possibility would cause more potential issues. Currently I'm slightly more leaning toward removing empty strings, just like you did :wink:. So, be fast and merge before I change my mind again and again 😝